app/src/app.tsx

```diff
@@ -59,6 +59,7 @@ export const App = (): React.ReactElement => {
     return null;
   };
   const [startDate, setStartDate] = React.useState<Date | null>(getUrlDate('date') || defaultDate);
+  const [startDateString, setStartDateString] = useUrlQueryState <string| null>('date', undefined, dateToString(defaultDate, 'yyyy-MM-dd'));
```

Comment:
when this pr is done, this should look like 
```
  const [startDate, setStartDate] = useDateUrlQueryState('date', 'yyyy-MM-dd', undefined, defaultDate);
```

and you should not need the line above (only one state) and not need `setUrlString`